src/main/java/com/xceptance/neodymium/common/browser/BrowserAfterRunner.java

```diff
@@ -0,0 +1,123 @@
+        // if we need to start new browser for the clean up and any browser configuration for the @After method
+        // was found, create a new driver
+        else if (startNewBrowserForCleanUp && browserConfiguration != null)
+        {
+            browserRunner.setUpTest();
+        }
+        else if (!isSuppressed && browserConfiguration == null)
+        {
+            throw new RuntimeException("No browser setting for " + (junit5 ? "@AfterEach" : "@After") + " method '" + (after.getName()) + "' was found. "
+                                       + "If browser was suppressed for the test but is annotated with @StartNewBrowserForCleanUp because browser isrequired for the clean up,"
+                                       + " please, use @Browser annotation to specify what browser is required for this clean up.");
+        }
+        boolean beforeFailed = false;
+        try
+        {
+            Throwable e = afterMethodInvocation.get();
+            if (e != null)
+            {
+                beforeFailed = true;
+                throw e;
+            }
+        }
+        finally
+        {
+            // if we did a set up of new driver before the @After method, we need to close it
+            if (startNewBrowserForCleanUp && browserConfiguration != null)
+            {
+                browserRunner.teardown(beforeFailed);
+                if (oldDriver != null)
+                {
+                    WebDriverRunner.setWebDriver(oldDriver.getWebDriver());
+                    Neodymium.setWebDriverStateContainer(oldDriver);
+                }
+                if (oldBrowserConfiguration != null)
+                {
+                    Neodymium.setBrowserProfileName(oldBrowserConfiguration.getConfigTag());
+                    Neodymium.setBrowserName(oldBrowserConfiguration.getCapabilities().getBrowserName());
+                }
+            }
+        }
+    }
+
+    private boolean shouldStartNewBrowser(Method each)
+    {
+        List<StartNewBrowserForCleanUp> methodStartNewBrowserForCleanUp = Data.getAnnotations(each,
+                                                                                              StartNewBrowserForCleanUp.class);
+        List<StartNewBrowserForCleanUp> classStartNewBrowserForCleanUp = Data.getAnnotations(each.getDeclaringClass(),
+                                                                                             StartNewBrowserForCleanUp.class);
+
+        // if global config for dontStartNewBrowserForCleanUp is set to false, we should not reach this point
+        boolean startNewBrowserForCleanUp = false;
+        if (!classStartNewBrowserForCleanUp.isEmpty())
+        {
+            startNewBrowserForCleanUp = true;
+        }
+
+        if (!methodStartNewBrowserForCleanUp.isEmpty())
+        {
+            startNewBrowserForCleanUp = true;
+        }
+
+        // if @After method is annotated with @SuppressBrowser annotation, no new browser should be started
```

Comment:
the code does not reflect what the comment demands here or did I see this wrong?